app/controllers/application_controller.rb

```diff
@@ -1,24 +1,11 @@
 class ApplicationController < ActionController::Base
   before_action :authenticate_user!
 
-  helper_method :current_user,
-                :logged_in?
-
-  private
-
-  def authenticate_user!
-    unless current_user
-      cookies[:user_id] = current_user&.id
-      cookies[:intended_url] = request.url
-      redirect_to login_path, alert: 'Are you a Guru? Verify your email and password!'
+  def after_sign_in_path_for(_tests)
```

Comment:
```suggestion
  def after_sign_in_path_for(user)
```
и далее использовать `user`, вместо `current_user`
такой вариант будет ближе всего к "родной" реализации девайса, т.е. меньше всего шансов, "что что-то пойдет не так"